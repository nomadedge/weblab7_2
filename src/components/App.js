import React, { Component } from 'react';

import LocalWeather from './LocalWeather';
import Favorites from './Favorites';

import '../styles/App.css';

class App extends Component {
    render() {
        return (
            <>
                <div id='local-weather'>
                    <h2>Local weather</h2>
                    <LocalWeather />
                </div>
                <div id='favorites'>
                    <h2>Your favorite cities</h2>
                    <Favorites />
                </div>
            </>
        );
    }
}

export default App;