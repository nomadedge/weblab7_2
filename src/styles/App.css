* {
    font-family: 'Segoe UI', sans-serif;
    margin: 0;
    padding: 0;
}

#local-weather{
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

#favorites{
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}