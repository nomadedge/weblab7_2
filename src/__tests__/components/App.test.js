import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from '../../components/App';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('App component', () => {
    test('matches the snapshot', () => {
        const localCity = {
            isFetching: true,
            city: null,
            weather: {},
            error: null
        };

        const existingArray = [
            {
                name: 'Kirov',
                isFetching: true,
                error: null,
                weather: {}
            },
            {
                name: 'Kiyv',
                isFetching: false,
                error: null,
                weather: {
                    iconUrl: "http://openweathermap.org/img/w/50n.png",
                    description: "mist",
                    temperature: "14",
                    pressure: 1022,
                    humidity: 100,
                    windSpeed: 3.6
                }
            }
        ];

        const store = mockStore({
            localWeather: {
                localCity
            },
            favoriteCities: {
                cities: existingArray
            }
        });

        const tree = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});