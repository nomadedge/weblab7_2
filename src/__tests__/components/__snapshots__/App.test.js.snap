// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
Array [
  <div
    id="local-weather"
  >
    <h2>
      Local weather
    </h2>
    <button
      onClick={[Function]}
    >
      Update local weather
    </button>
    <div
      id="weather-container"
    >
      <div />
    </div>
  </div>,
  <div
    id="favorites"
  >
    <h2>
      Your favorite cities
    </h2>
    <div
      id="cityAdd"
    >
      <form
        name="cityAdd"
        onSubmit={[Function]}
      >
        <input
          id="input"
          placeholder="Enter new favorite city"
        />
        <input
          type="submit"
          value="Add city"
        />
      </form>
    </div>
    <div
      id="favorite"
    >
      <div
        id="weather-container"
      >
        <div
          id="left-block"
        >
          <div
            id="city-name"
          >
            Kirov
          </div>
        </div>
        <div
          id="right-block"
        >
          <div
            id="spinner"
          />
        </div>
      </div>
    </div>
    <div
      id="favorite"
    >
      <button
        id="delete-button"
        onClick={[Function]}
      >
        Delete city
      </button>
      <div
        id="weather-container"
      >
        <div
          id="left-block"
        >
          <div
            id="city-name"
          >
            Kiyv
          </div>
          <div
            id="icon-temp"
          >
            <img
              alt="Weather icon"
              src="http://openweathermap.org/img/w/50n.png"
            />
            <div
              id="temp"
            >
              14
              Â°C
            </div>
          </div>
        </div>
        <div
          id="right-block"
        >
          <div
            className="info"
          >
            <div>
              Description:
            </div>
            <div
              className="value"
            >
              mist
            </div>
          </div>
          <div
            className="info"
          >
            <div>
              Barometer:
            </div>
            <div
              className="value"
            >
              1022
               hPa
            </div>
          </div>
          <div
            className="info"
          >
            <div>
              Humidity:
            </div>
            <div
              className="value"
            >
              100
               %
            </div>
          </div>
          <div
            className="info"
          >
            <div>
              Wind:
            </div>
            <div
              className="value"
            >
              3.6
               m/s
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
